Cockroach -> CockroachDB

tbg opened the pull request.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/15582)
<!-- Reviewable:end -->

knz commented on pkg/cli/start.go:
```diff
@@ -343,7 +343,7 @@ func runStart(cmd *cobra.Command, args []string) error {
 			s, err = server.NewServer(serverCfg, stopper)
 			if err != nil {
-				return errors.Wrap(err, "failed to start Cockroach server")
+				return errors.Wrap(err, "failed to start CockroachDB server")
```
`Wrapf(err, "failed to start %s", os.Argv[1])`

That's the more correct way

tschottdorf replied:
I don't think this message should depend on the binary name. In the interest of not bike shedding about this, are you OK just dropping the word `Cockroach` instead? It doesn't add anything - you know what you're trying to run:

```
$ ./cockroach start
*
* ERROR: failed to start Cockroach server: problem using security settings, did you mean to use --insecure?: no CA certificate found
*
Failed running "start"
```

knz commented:
Yes OK +1
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

tbg committed:
Remove spurious Cockroach

Doesn't make sense to mention in this error, and if so, it would have to be the
full name CockroachDB. Either way, removing it here.
